tests-dynamic.scm

```diff
@@ -164,15 +165,50 @@
         '(if (null? xs) ys
              (cons (car xs)
                    (append (cdr xs) ys))))))
+(test
```

Comment:
It would be useful to have a comment stating what we're trying to find out with these tests.